sympy/solvers/ode/tests/test_single.py

```diff
@@ -1517,17 +1669,42 @@ def _get_examples_ode_sol_separable():
 
     'separable_21': {
         'eq': f(x)*diff(f(x), x) + x - 3*x*f(x)**2,
-        'sol': [
-            Eq(f(x), -sqrt(3)*sqrt(C1*exp(3*x**2) + 1)/3),
-            Eq(f(x), sqrt(3)*sqrt(C1*exp(3*x**2) + 1)/3)
-        ],
+        'sol': [Eq(f(x), -sqrt(3)*sqrt(C1*exp(3*x**2) + 1)/3),
+        Eq(f(x), sqrt(3)*sqrt(C1*exp(3*x**2) + 1)/3)],
     },
 
     'separable_22': {
         'eq': f(x).diff(x) - exp(x + f(x)),
         'sol': [Eq(f(x), log(-1/(C1 + exp(x))))],
         'XFAIL': ['lie_group'] #It shows 'NoneType' object is not subscriptable for lie_group.
     },
+
+    # https://github.com/sympy/sympy/issues/7081
+    'separable_23': {
+        'eq': x*(f(x).diff(x)) + 1 - f(x)**2,
+        'sol': [Eq(f(x), -1/(-C1 + x**2)*(C1 + x**2))],
+    },
+
+    # # https://github.com/sympy/sympy/issues/10379
+    # 'separable_24': {
+    #     'eq': f(t).diff(t)-(1-51.05*y*f(t)),
+    #     'sol': [Eq(f(t), (0.019588638589618*exp(y*(C1 - 51.05*t)) + 0.019588638589618)/y)],
```

Comment:
The solution should be:
```python
Eq(f(t), (0.019588638589618023*exp(y*(C1 - 51.049999999999997*t)) + 0.019588638589618023)/y)
```
Compare the difference between `repr` and `srepr`:
```python
In [11]: sol = dsolve(f(t).diff(t)-(1-51.05*y*f(t)))

In [12]: repr(sol)
Out[12]: 'Eq(f(t), (0.019588638589618*exp(y*(C1 - 51.05*t)) + 0.019588638589618)/y)'

In [13]: srepr(sol)
Out[13]: "Equality(Function('f')(Symbol('t')), Mul(Pow(Symbol('y'), Integer(-1)), Add(Mul(Float('0.019588638589618023', precision=53), exp(Mul(Symbol('y'), Add(Symbol('C1'), Mul(Integer(-1), Float('51.049999999999997', precision=53), Symbol('t')))))), Float('0.019588638589618023', precision=53))))"
```

Reply:
> The solution should be:
> 
> ```python
> Eq(f(t), (0.019588638589618023*exp(y*(C1 - 51.049999999999997*t)) + 0.019588638589618023)/y)
> ```
expected solution is defined as list throughout the file so that dsolve can be tested for missing solutions too.
Also I couldn't understand how repr and srepr will impact this? if you could explain a bit more.

Reply:
Yes, it should be a list.

The `repr` doesn't show enough digits to give the exact same float.

Reply:
so should I keep this as expected solution `[Eq(f(t), (0.019588638589618023*exp(y*(C1 - 51.049999999999997*t)) + 0.019588638589618023)/y)]` ?

Reply:
This also fails.

Reply:
How does it fail?

It works fine here:
```python
In [114]: eq = f(t).diff(t)-(1-51.05*y*f(t))

In [115]: sol = Eq(f(t), (0.019588638589618023*exp(y*(C1 - 51.049999999999997*t)) + 0.019588638589618023)/y)

In [116]: dsolve(eq) == sol
Out[116]: True

In [117]: checkodesol(eq, sol)
Out[117]: (True, 0)
```

Reply:
sorry my bad, I was explicity checking with srepr also.